app/sessionStore.js

```diff
@@ -845,11 +845,17 @@ module.exports.runPreMigrations = (data) => {
       if (data.cache) {
         delete data.cache.bookmarkLocation
       }
+    }
+
+    // pinned top sites were stored in the wrong position in 0.19.x
+    // and on some updates ranging from 0.20.x/0.21.x
+    // allowing duplicated items. See #12941
+    let pinnedTopSitesCleanup = false
+    try {
+      pinnedTopSitesCleanup = compareVersions(data.lastAppVersion, '0.22.00') < 1
```

Comment:
should we have another check for this, so it only runs once? Specifically we could capture this in the state in an area called "migrations" (which was removed with https://github.com/brave/browser-laptop/commit/ccee1223cbe565c546b6f103b661ae749bc85971#diff-b4e400e260f324745d12616a8c9c1fc6L377 since nothing else is using it).  You could basically store the timestamp this was ran on and default the value to undefined

Reply:
Also, note to self: we'll want to update this version before we ship. We might want to create an issue to track that